src/functions.py

```diff
@@ -23,7 +23,7 @@ def _get_py_capsule(graph):
 from .VertexPartition import *
 from .Optimiser import *
 
-def find_partition(graph, partition_type, initial_membership=None, weights=None, n_iterations=2, seed=None, **kwargs):
+def find_partition(graph, partition_type, initial_membership=None, weights=None, n_iterations=2, seed=None, max_comm_size=0, **kwargs):
```

Comment:
No, I think you were correct above. Using `max_comm_size` is rare enough that I shouldn't have inflicted it on the API of all these functions. I took it out here as well.

Reply:
Correction: It _does_ make sense to pass `max_com_size` to these functions since they create an internal `Optimiser` and there's no easy way to tweak it short of an explicit parameter. That said, I added an explicit check for negative values to the code.

Reply:
The same hence goes for other settings (e.g. `consider_empty_communtiy`), but as said, I think we can allow an exception for this case.